x-pack/plugins/apm/common/rules/schema.ts

```diff
@@ -62,6 +62,12 @@ export const anomalyParamsSchema = schema.object({
     schema.literal(ML_ANOMALY_SEVERITY.MINOR),
     schema.literal(ML_ANOMALY_SEVERITY.WARNING),
   ]),
+  anomalyDetectorType: schema.oneOf([
+    schema.literal(ANOMALY_DETECTOR_TYPES.ALL),
```

Comment:
We should not introduce an "All" option here. The only reason we did this for `service.name` and `service.environment` was to make the rules open-ended, so they would apply to future services/environments so users didn't have to update their rules every time they added new services). That doesn't apply to detectors, as we don't expect to add new detectors frequently and if we do, we most likely don't want them to apply to existing jobs.

Most importantly, the "All" option came at a costs and has resulted in many bugs. We should not make the same mistakes  by adding the All option for detector types too. Instead, we should capture a list of selected detectors. I believe this will both be simpler, and will allow us to modify the rule (now or in the future) so that users can cherry-pick multiple detector types, say latency and throughput but leave out failed transaction rate detector.

Reply:
@sqren I'll put this PR back to draft, and for the change you are suggesting, I think we will need product/design to have a look, AFAIK there's no other rule creation with checkboxes, so I would like for @boriskirov help to decide how to integrate this.

Reply:
Is it not just a matter of changing the select (single-select) to a checkbox (multi-select)? I don't believe this is a big change.